server/sonar-server/src/main/java/org/sonar/server/es/BulkIndexer.java

```diff
@@ -112,35 +178,17 @@ public BulkIndexer setFlushByteSize(long flushByteSize) {
   @Override
   public void start() {
     Preconditions.checkState(bulkRequest == null, ALREADY_STARTED_MESSAGE);
-    if (size == Size.LARGE) {
-      largeInitialSettings = Maps.newHashMap();
-      Map<String, Object> bulkSettings = Maps.newHashMap();
-      GetSettingsResponse settingsResp = client.nativeClient().admin().indices().prepareGetSettings(indexName).get();
-
-      // deactivate replicas
-      int initialReplicas = Integer.parseInt(settingsResp.getSetting(indexName, IndexMetaData.SETTING_NUMBER_OF_REPLICAS));
-      if (initialReplicas > 0) {
-        largeInitialSettings.put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, initialReplicas);
-        bulkSettings.put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0);
-      }
-
-      // deactivate periodical refresh
-      String refreshInterval = settingsResp.getSetting(indexName, REFRESH_INTERVAL_SETTING);
-      largeInitialSettings.put(REFRESH_INTERVAL_SETTING, refreshInterval);
-      bulkSettings.put(REFRESH_INTERVAL_SETTING, "-1");
-
-      updateSettings(bulkSettings);
-    }
-    bulkRequest = client.prepareBulk().setRefresh(false);
+    size.beforeStart(this);
+    bulkRequest = client.prepareBulkProcessor(new BulkProcessorListener())
```

Comment:
What about explicitly set `.setBackoffPolicy(BackoffPolicy.exponentialBackoff())` because it's the expected behavior ? It would allow to detect potential changes in next upgrades of Elasticsearch.